### PR TITLE
Add unit testing for adq-patch (local matching)

### PR DESCRIPTION
**Pull request by flomnes**

### CODE REVIEW

File: .github/workflows/ubuntu-system.yml [hunk 1]
```diff
@@ -86,6 +86,11 @@ jobs:
            cd _build
            ctest -C Release --output-on-failure -R unfeasible
 
+    - name: Run adq-patch related tests
+      run: |
+           cd build
```
Comment (flomnes) [inline .github/workflows/ubuntu-system.yml]:
```suggestion
           cd _build
```

File: .github/workflows/windows-vcpkg.yml [hunk 2]
```diff
@@ -116,6 +116,12 @@ jobs:
            cd _build
            ctest -C Release --output-on-failure -R unfeasible
 
+    - name: Run adq-patch related tests
+      run: |
+           cd build
```
Comment (flomnes) [inline .github/workflows/windows-vcpkg.yml]:
```suggestion
           cd _build
```

File: src/tests/CMakeLists.txt [hunk 3]
```diff
@@ -10,11 +10,18 @@ if (Boost_FOUND)
 endif()
 
 
-add_subdirectory(src)
 
-#end to end test require boost 1.6.x because for boost::test_tools::tolerance 
-if(Boost_VERSION VERSION_GREATER 106000)
+
+
+# end to end test require boost 1.6.x because for boost::test_tools::tolerance
```
Comment (guilpier-code) [inline src/tests/CMakeLists.txt:16]: In the comment, **1.6.x** seems incorrect, better replace it with **1.60.x**

File: src/tests/CMakeLists.txt [hunk 4]
```diff
@@ -10,11 +10,18 @@ if (Boost_FOUND)
...
+# old versions of Boost don't contain a '.', also ignore them
+# example : Boost 1.53.0 returns 105300 as Boost_VERSION
+string(FIND ${Boost_VERSION} . RESULT_BOOST)
+if ((${Boost_VERSION} VERSION_GREATER_EQUAL 1.60.0) AND NOT (${RESULT_BOOST} EQUAL -1))
```
Comment (guilpier-code) [inline src/tests/CMakeLists.txt:20]: On the CI, at least on Windows, I can see that Boost_VERSION is 107900, not 1.79.00. The consequences is that some tests are not generated.

File: .github/workflows/windows-vcpkg.yml [hunk 5]
```diff
@@ -116,6 +116,12 @@ jobs:
            cd _build
            ctest -C Release --output-on-failure -R unfeasible
 
+    - name: Run adq-patch related tests
```
Comment (guilpier-code) [inline .github/workflows/windows-vcpkg.yml:119]: At some point in the life of the workflow **yaml** file, some tests were run : tests on class matrix, on ini files, on TS numbers attribution, etc. 
They are not run anymore on Windows, as well as on Ubuntu. We should run them again.

File: src/tests/src/solver/optimisation/adequacy_patch.cpp [hunk 6]
```diff
@@ -0,0 +1,160 @@
...
+                                       bool SetNTCOutsideToOutsideToZero,
+                                       bool SetNTCOutsideToInsideToZero)
+{
+  PROBLEME_HEBDO problem;
+  problem.adequacyPatchRuntimeData.originAreaType.resize(1);
+  problem.adequacyPatchRuntimeData.extremityAreaType.resize(1);
+
+  problem.adequacyPatchRuntimeData.originAreaType[0] = originType;
+  problem.adequacyPatchRuntimeData.extremityAreaType[0] = extremityType;
+  problem.adqPatch = std::unique_ptr<AdequacyPatchParameters>(new AdequacyPatchParameters());
```
Comment (guilpier-code) [inline src/tests/src/solver/optimisation/adequacy_patch.cpp]: `problem.adqPatch` should be renamed `problem.adqPatchParams`

File: src/tests/src/solver/optimisation/adequacy_patch.cpp [hunk 7]
```diff
@@ -0,0 +1,160 @@
...
+static double origineExtremite = -1;
+static double extremiteOrigine = 5;
+
+using namespace Antares::Data::AdequacyPatch;
+
+// NOTE
+// Xmax limits the flux origin -> extremity (direct)
+// -Xmin limits the flux extremity -> origin (indirect)
+
+std::pair<double, double> one_timestep(AdequacyPatchMode originType,
```
Comment (guilpier-code) [inline src/tests/src/solver/optimisation/adequacy_patch.cpp]: **one_timestep** function should be renamed **setNTCboundsForOneTimeStep** I suppose

**Review by guilpier-code — CHANGES_REQUESTED** [review]

File: src/tests/CMakeLists.txt (hunk 3 above)
Comment (flomnes) [reply src/tests/CMakeLists.txt:16]: Corrected on **develop**

File: src/tests/CMakeLists.txt (hunk 4 above)
Comment (flomnes) [reply src/tests/CMakeLists.txt:20]: Corrected on **develop**

File: .github/workflows/windows-vcpkg.yml (hunk 5 above)
Comment (flomnes) [reply .github/workflows/windows-vcpkg.yml:119]: Corrected on **develop**

File: src/tests/src/solver/optimisation/adequacy_patch.cpp (hunk 6 above)
Comment (flomnes) [reply src/tests/src/solver/optimisation/adequacy_patch.cpp]: Fixed

File: src/tests/src/solver/optimisation/adequacy_patch.cpp (hunk 7 above)
Comment (flomnes) [reply src/tests/src/solver/optimisation/adequacy_patch.cpp]: Done

**Pull request merged**